android/java/org/chromium/chrome/browser/search_engines/settings/BraveSearchEngineAdapter.java

```diff
@@ -114,4 +151,12 @@ public View getView(int position, View convertView, ViewGroup parent) {
         }
         return view;
     }
+
+    // Wrapper for setting profile when running BraveTemplateUrlServiceFactory.get()
+    // No need to synchronize because BraveTemplateUrlServiceFactory shall always be on main UI thread
+    private void runTemplateUrlServiceWithProfile(Runnable r) {
+        BraveTemplateUrlServiceFactory.setCurrentProfile(mProfile);
+        r.run();
+        BraveTemplateUrlServiceFactory.setCurrentProfile(null);
```

Comment:
not sure I understand what we do here

Reply:
So what happens is the original call in chromium doesn't have a parameter, and bytecode cannot rewrite calls to add it. So I use a workaround that I add a static field in the `BraveTemplateUrlServiceFactory` class and set it before any function call to it. Otherwise we need to overwrite every function in chromium's `SearchEngineAdapter.java`.

Reply:
why do we need to call it twice, one time before the `r.run()` with a profile and another one after that?

Reply:
I don't see as well how `sProfile` is used, do we need it at all?

Reply:
> why do we need to call it twice, one time before the r.run() with a profile and another one after that?

It is once for setting the static field and once for clearing it so other calls to `get()` doesn't get stuck with the profile. I accidentally deleted the `sProfile` part when doing the other fix for getting tab models in native, thanks for pointing that out.

Reply:
Fixed now.

Reply:
ok, got it, could you please make another pass and make sure everything works as intended before the merge?

Reply:
Yeah I'll rebase, fix the formatting etc. and check everything is fine.